great_expectations/checkpoint/v1_checkpoint.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, List, Union, cast
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations import project_manager
+from great_expectations._docs_decorators import public_api
+from great_expectations.checkpoint.actions import ValidationAction  # noqa: TCH001
+from great_expectations.compatibility.pydantic import BaseModel, validator
+from great_expectations.core.serdes import _IdentifierBundle
+from great_expectations.core.validation_config import ValidationConfig
+from great_expectations.render.renderer.renderer import Renderer
+
+if TYPE_CHECKING:
+    from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+    from great_expectations.data_context.store.validation_config_store import (
+        ValidationConfigStore,
+    )
+
+
+class Checkpoint(BaseModel):
+    """
+    A Checkpoint is the primary means for validating data in a production deployment of Great Expectations.
+
+    Checkpoints provide a convenient abstraction for running a number of validations and triggering a set of actions
+    to be taken after the validation step.
+
+    Args:
+        validations: List of validation configs to be run.
+        actions: List of actions to be taken after the validations are run.
+
+    """
+
+    name: str
+    validations: List[ValidationConfig]
```

Comment:
Why not a list of `ValidationDefinitions` instead of config objects? That's what this shows: https://greatexpectations.atlassian.net/wiki/spaces/SUP/pages/917471267/Validation+Workflows

Or does the ValidationConfig need to be renamed?

Reply:
Oh, I think ValidationConfig does need to be renamed. That can be done in a separate PR.

Reply:
Needs to be renamed!